r/tests/testthat/test-Table.R

```diff
@@ -334,5 +334,5 @@ test_that("Table metadata", {
 
 test_that("Table handles null type (ARROW-7064)", {
   tab <- Table$create(a = 1:10, n = vctrs::unspecified(10))
-  expect_equal(tab$schema,  schema(a = int32(), n = null()))
+  expect_true(tab$schema$Equals(schema(a = int32(), n = null()), FALSE))
```

Comment:
Maybe `vctrs::unspecified()` can be special cased so that it does not generate additional info in the schema (it's just the class) here. 

Or maybe the equality operator could disregard the metadata altogether (i.e. changing the default).

Reply:
I think you can use `expect_equivalent()` instead of `expect_equal` and it will skip the metadata comparison; cf. https://github.com/apache/arrow/blob/master/r/R/arrow-package.R#L111 

(of course, we'll have to fix this for the new version of `testthat` that doesn't use all.equal under the hood)

Reply:
Thanks.